### PR TITLE
catppuccin-catwalk: init at 0.4.0

### PR DESCRIPTION
**Pull request by ryanccn**
###### Description of changes

https://github.com/catppuccin/toolbox

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by getchoo** [comment]
Result of `nixpkgs-review pr 245476` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>catppuccin-catwalk (python310Packages.catppuccin-catwalk)</li>
    <li>catppuccin-catwalk.dist (python310Packages.catppuccin-catwalk.dist)</li>
    <li>python311Packages.catppuccin-catwalk</li>
    <li>python311Packages.catppuccin-catwalk.dist</li>
  </ul>
</details>

**Review by getchoo — APPROVED** [review]

File: pkgs/development/python-modules/catppuccin-catwalk/default.nix [hunk 1]
```diff
@@ -0,0 +1,29 @@
+{
+  lib,
+  buildPythonPackage,
+  fetchPypi,
+  pillow,
+}:
+buildPythonPackage rec {
```
Comment (natsukium) [inline pkgs/development/python-modules/catppuccin-catwalk/default.nix:7]:
```suggestion
buildPythonApplication rec {
```
I don't think this is a python library.
Would you please use `buildPythonApplication` here?

File: pkgs/development/python-modules/catppuccin-catwalk/default.nix [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  pname = "catppuccin-catwalk";
+  version = "0.4.0";
```
Comment (natsukium) [inline pkgs/development/python-modules/catppuccin-catwalk/default.nix:9]:
```suggestion
  version = "0.4.0";
  format = "pyproject";
```
Add `format`, please.

File: pkgs/development/python-modules/catppuccin-catwalk/default.nix [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+  src = fetchPypi {
+    inherit version;
+    pname = "catppuccin_catwalk";
+    sha256 = "sha256-5TAw5H3soxe9vLhfj1qs8uMr4ybrHlCj4zdsMzvPo6s=";
```
Comment (natsukium) [inline pkgs/development/python-modules/catppuccin-catwalk/default.nix:14]:
```suggestion
    hash = "sha256-5TAw5H3soxe9vLhfj1qs8uMr4ybrHlCj4zdsMzvPo6s=";
```
We should use SRI hash.
https://nixos.org/manual/nixpkgs/stable/#fetchurl

File: pkgs/development/python-modules/catppuccin-catwalk/default.nix [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+  };
+
+  propagatedBuildInputs = [
+    pillow
+  ];
+
+  doCheck = false;
```
Comment (natsukium) [inline pkgs/development/python-modules/catppuccin-catwalk/default.nix:21]: Would you please add the comment about why set `doCheck = false`?

File: pkgs/development/python-modules/catppuccin-catwalk/default.nix [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+
+  meta = with lib; {
+    homepage = "https://github.com/catppuccin/toolbox";
+    description = "A CLI for Catppuccin that takes in four showcase images and displays them all at once.";
```
Comment (natsukium) [inline pkgs/development/python-modules/catppuccin-catwalk/default.nix:25]:
```suggestion
    description = "A CLI for Catppuccin that takes in four showcase images and displays them all at once";
```
Please remove a period at the end.
https://nixos.org/manual/nixpkgs/stable/#var-meta-description

**Review by natsukium — COMMENTED** [review]
Could you please reformat with `nixpkgs-fmt`?

File: pkgs/top-level/all-packages.nix [hunk 6]
```diff
@@ -411,6 +411,8 @@ with pkgs;
 
   catatonit = callPackage ../applications/virtualization/catatonit { };
 
+  catppuccin-catwalk = with python3Packages; toPythonApplication catppuccin-catwalk;
```
Comment (natsukium) [inline pkgs/top-level/all-packages.nix:414]:
```suggestion
  catppuccin-catwalk = callPackage ../development/tools/misc/catppuccin-catwalk { };
```
I'm not sure if this is the right place, but please move it from python-module.

File: pkgs/development/python-modules/catppuccin-catwalk/default.nix [hunk 7]
```diff
@@ -0,0 +1,35 @@
+{ lib
+, buildPythonApplication
+, fetchPypi
+, pillow
+, poetry-core
+,
+}: buildPythonApplication rec {
```
Comment (natsukium) [inline pkgs/development/python-modules/catppuccin-catwalk/default.nix:7]:
```suggestion
{ lib
, fetchPypi
, python3
}:

python3.pkgs.buildPythonApplication rec {
```

File: pkgs/development/python-modules/catppuccin-catwalk/default.nix [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+  version = "0.4.0";
+  format = "pyproject";
+
+  src = fetchPypi {
+    inherit version;
+    pname = "catppuccin_catwalk";
+    hash = "sha256-5TAw5H3soxe9vLhfj1qs8uMr4ybrHlCj4zdsMzvPo6s=";
+  };
+
+  propagatedBuildInputs = [
```
Comment (natsukium) [inline pkgs/development/python-modules/catppuccin-catwalk/default.nix:18]:
```suggestion
  propagatedBuildInputs = with python3.pkgs; [
```

File: pkgs/development/python-modules/catppuccin-catwalk/default.nix [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+    pillow
+  ];
+
+  nativeBuildInputs = [
```
Comment (natsukium) [inline pkgs/development/python-modules/catppuccin-catwalk/default.nix:22]:
```suggestion
  nativeBuildInputs = with python3.pkgs; [
```

File: pkgs/development/python-modules/catppuccin-catwalk/default.nix [hunk 10]
```diff
@@ -0,0 +1,35 @@
...
+    poetry-core
+  ];
+
+  # upstream does not have testing
+  doCheck = false;
+
```
Comment (natsukium) [inline pkgs/development/python-modules/catppuccin-catwalk/default.nix:28]:
```suggestion

  pythonImports = [
    "catwalk"
  ];

```
This is not a library, but `pythonImportsCheck` should provide a simple test.

File: pkgs/development/tools/misc/catppuccin-catwalk/default.nix [hunk 11]
```diff
@@ -0,0 +1,34 @@
+{ lib
+, fetchPypi
+, python3
+,
+}: python3.pkgs.buildPythonApplication rec {
```
Comment (natsukium) [inline pkgs/development/tools/misc/catppuccin-catwalk/default.nix:5]:
```suggestion
}:

python3.pkgs.buildPythonApplication rec {
```

File: pkgs/development/tools/misc/catppuccin-catwalk/default.nix [hunk 12]
```diff
@@ -0,0 +1,34 @@
...
+    poetry-core
+  ];
+
+  pythonImports = [
+    "catwalk"
+  ];
+
+  meta = with lib; {
+    homepage = "https://github.com/catppuccin/toolbox";
+    description = "A CLI for Catppuccin that takes in four showcase images and displays them all at once.";
```
Comment (natsukium) [inline pkgs/development/tools/misc/catppuccin-catwalk/default.nix:30]:
```suggestion
    description = "A CLI for Catppuccin that takes in four showcase images and displays them all at once";
```

File: pkgs/development/tools/misc/catppuccin-catwalk/default.nix [hunk 13]
```diff
@@ -0,0 +1,34 @@
...
+    hash = "sha256-5TAw5H3soxe9vLhfj1qs8uMr4ybrHlCj4zdsMzvPo6s=";
+  };
+
+  propagatedBuildInputs = with python3.pkgs; [
+    pillow
+  ];
+
+  nativeBuildInputs = with python3.pkgs; [
+    poetry-core
+  ];
```
Comment (natsukium) [inline pkgs/development/tools/misc/catppuccin-catwalk/default.nix:22]:
```suggestion
  nativeBuildInputs = with python3.pkgs; [
    poetry-core
  ];

  propagatedBuildInputs = with python3.pkgs; [
    pillow
  ];
```
Usually, `nativeBuildInputs` come first.

**Review by natsukium — COMMENTED** [review]
nits
besides these, LGTM.

File: pkgs/development/tools/misc/catppuccin-catwalk/default.nix [hunk 14]
```diff
@@ -0,0 +1,36 @@
+{ lib
+, fetchPypi
+, python3
+,
+}:
+
+python3.pkgs.buildPythonApplication rec {
+  pname = "catppuccin-catwalk";
```
Comment (drupol) [inline pkgs/development/tools/misc/catppuccin-catwalk/default.nix:8]: The project is https://github.com/catppuccin/toolbox, why naming it `catwalk`?

Comment (natsukium) [reply pkgs/development/tools/misc/catppuccin-catwalk/default.nix:8]: `catwalk` is one of the `toolbox` packages.
https://github.com/catppuccin/toolbox#-catwalk

Comment (drupol) [reply pkgs/development/tools/misc/catppuccin-catwalk/default.nix:8]: Thanks!

**Review by drupol — APPROVED** [review]

**Pull request merged**